OsmAnd/src/net/osmand/plus/views/layers/MapTileLayer.java

```diff
@@ -74,21 +89,37 @@ public void initLayer(@NonNull OsmandMapTileView view) {
 
 		useSampling = Build.VERSION.SDK_INT < 28;
 
-		paintBitmap = new Paint();
-		paintBitmap.setFilterBitmap(true);
-		paintBitmap.setAlpha(getAlpha());
-		
+		if (view.getMapRenderer() == null || paintBitmap == null) {
+			paintBitmap = new Paint();
+			paintBitmap.setFilterBitmap(true);
+			paintBitmap.setAlpha(getAlpha());
+		}
+
 		if (mapTileAdapter != null) {
 			mapTileAdapter.initLayerAdapter(this, view);
 		}
 	}
-	
+
 	@Override
 	public void setAlpha(int alpha) {
+		if (getAlpha() == alpha){
+			return;
+		}
```

Comment:
**super** must be called anyway, formatting

Reply:
Got it.